tests/test_job.py

```diff
@@ -127,30 +157,41 @@ def test_run_to_completion_success(self, mock_get_status, mock_create_namespaced
     @patch("tesk_core.job.Job.get_status", side_effect=[("Running", True)])
     def test_run_to_completion_cancelled(self, mock_get_status, mock_create_namespaced_job, mock_check_cancelled,
                                mock_job_delete):
+        """
+        Checking if the Job is cancelled
+        """
         for executor in self.data['executors']:
             jobname = executor['metadata']['name']
             job = Job(executor, jobname, taskmaster.args.namespace)
             status = job.run_to_completion(taskmaster.args.poll_interval, taskmaster.check_cancelled,
                                            taskmaster.args.pod_timeout)
             self.assertEqual(status, "Cancelled")
 
-    @patch("kubernetes.client.CoreV1Api.list_namespaced_pod", side_effect=list_namespaced_pod_error)
+    @patch("kubernetes.client.CoreV1Api.list_namespaced_pod")
     @patch("kubernetes.client.BatchV1Api.read_namespaced_job", side_effect=read_namespaced_job_error)
     @patch("tesk_core.job.Job.delete")
     @patch("tesk_core.taskmaster.check_cancelled", return_value=False)
     @patch("kubernetes.client.BatchV1Api.create_namespaced_job")
     def test_run_to_completion_error(self, mock_create_namespaced_job, mock_check_cancelled,
-                                         mock_job_delete,mock_list_namespaced_pod,  mock_read_namespaced_job):
+                                         mock_job_delete,mock_read_namespaced_job,mock_list_namespaced_pod  ):
+        """
+        Testing if the Job is in pending state when the status of the pod is pending
+        state and reason is ImagePullBackOff
+        """
```

Comment:
Right, but that comment is confusing, too.
The method under test is `job.run_to_completion` and asserted result is `Error` and you write:
`Testing if the Job is in pending state` - actually you assert that jo.run_to_completion will return error, right? You do not check, if any Job is in pending state.